pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,191 @@
+package describe
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/golang/glog"
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/encoding/dot"
+	"github.com/gonum/graph/path"
+	"github.com/gonum/graph/traverse"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildgraph "github.com/openshift/origin/pkg/build/graph/nodes"
+	"github.com/openshift/origin/pkg/client"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// NotFoundErr is returned when the imageStreamTag (ist) of interest cannot
+// be found in the graph. This doesn't mean though that the ist does not
+// exist. A user may have an image stream without a build configuration
+// pointing at it. In that case, the ist of interest simply doesn't have
+// other dependant ists
+type NotFoundErr string
+
+func (e NotFoundErr) Error() string {
+	return fmt.Sprintf("couldn't find image stream tag: %q", string(e))
+}
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	namespaces []string
+	output     string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, namespaces []string, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, namespaces: namespaces, output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace(s)
+func (d *ChainDescriber) MakeGraph(namespaces []string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{}
+	for _, namespace := range namespaces {
+		glog.V(4).Infof("Loading build configurations from %q", namespace)
+		loaders = append(loaders, &bcLoader{namespace: namespace, lister: d.Interface})
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
```

Comment:
Explain why namespace is a separate param.  I agree that it should be separate, but explain it to future maintainers.